docker-versions-maven-plugin/src/main/java/com/github/cafapi/docker_versions/plugins/IgnoreVersionsHelper.java

```diff
@@ -0,0 +1,111 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class for managing ignore versions.
+ */
+public final class IgnoreVersionsHelper
+{
+    private static final List<String> DEFAULT_IGNORE_VERSIONS = Arrays.asList(new String[]{"latest", "stable"});
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IgnoreVersionsHelper.class);
+
+    public static List<String> getRelevantTags(
+        final Set<IgnoreVersion> configuredIgnoreVersions,
+        final List<String> tags,
+        final String repository)
+    {
+        LOGGER.debug("Configured ignore versions: {}", configuredIgnoreVersions);
+        final Map<String, Set<IgnoreVersion>> imageIgnoreVersionsLookup = new HashMap<>();
+        return tags.stream()
+            .filter(t -> !isIgnoredVersion(configuredIgnoreVersions, imageIgnoreVersionsLookup, repository, t))
+            .collect(Collectors.toList());
+    }
+
+    private static boolean isIgnoredVersion(
+        final Set<IgnoreVersion> configuredIgnoreVersions,
+        final Map<String, Set<IgnoreVersion>> imageIgnoreVersionsLookup,
+        final String imageName,
+        final String tag)
+    {
+        if (DEFAULT_IGNORE_VERSIONS.contains(tag.toLowerCase(Locale.ENGLISH))) {
+            return true;
+        }
+
+        if (configuredIgnoreVersions.isEmpty()) {
+            return false;
+        }
+
+        final Set<IgnoreVersion> imageIgnoreVersions
+            = getImageIgnoreVersions(configuredIgnoreVersions, imageName, imageIgnoreVersionsLookup);
+
+        boolean isMatch;
+        for (final IgnoreVersion iVersion : imageIgnoreVersions) {
+            isMatch = "regex".equals(iVersion.getType())
+                ? Pattern.matches(iVersion.getVersion(), tag)
+                : iVersion.getVersion().equals(tag);
+            LOGGER.trace("Check if tag {} matches version pattern {} : {}", tag, iVersion.getVersion(), isMatch);
+            if (isMatch) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static Set<IgnoreVersion> getImageIgnoreVersions(
+        final Set<IgnoreVersion> configuredIgnoreVersions,
+        final String imageName,
+        final Map<String, Set<IgnoreVersion>> imageIgnoreVersionsLookup)
+    {
+        if (imageIgnoreVersionsLookup.containsKey(imageName)) {
+            return imageIgnoreVersionsLookup.get(imageName);
+        }
+
+        final Set<IgnoreVersion> imageIgnoreVersions = new HashSet<>();
+
+        for (final IgnoreVersion iVersion : configuredIgnoreVersions) {
+            final Set<String> images = iVersion.getImages();
+            if (images == null || images.isEmpty()) {
+                imageIgnoreVersions.add(iVersion);
+                continue;
+            }
+            for (final String img : images) {
+                LOGGER.trace("Check if image {} matches pattern {}", imageName, img);
+                if (Pattern.matches(img, imageName)) {
```

Comment:
Is there any way to compile the pattern once rather than each time it is needed?